### PR TITLE
migrate tasks

### PR DESCRIPTION
**Pull request by chkeita**
Migrate task function to C#.

- [x] check-pr passed

### CODE REVIEW

File: src/ApiService/ApiService/Functions/Tasks.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        var request = await RequestHandling.ParseRequest<TaskSearch>(req);
+        if (!request.IsOk) {
+            return await _context.RequestHandling.NotOk(req, request.ErrorV, "task get");
+        }
+
+        if (request.OkV.TaskId != null) {
+            var task = await _context.TaskOperations.GetByTaskId(request.OkV.TaskId.Value);
+            if (task == null) {
+                return await _context.RequestHandling.NotOk(req, new Error(ErrorCode.INVALID_REQUEST, new[] { "unable to find task"
+                }), "pool get");
```
Comment (stishkin) [inline src/ApiService/ApiService/Functions/Tasks.cs:38]: task get ?

File: src/ApiService/ApiService/onefuzzlib/Config.cs [hunk 2]
```diff
@@ -257,4 +264,257 @@ await _containers.GetContainerSasUrl(x.Item2.Name, StorageType.Corpus, ConvertPe
...
+
+        if (definition.Features.Contains(TaskFeature.TargetMustUseInput) && !TargetUsesInput(config)) {
+            return ResultVoid<TaskConfigError>.Error(new TaskConfigError("{input} must be used in target_env or target_options"));
+        }
+
+        if (config.Vm != null) {
+            return ResultVoid<TaskConfigError>.Error(new TaskConfigError("specifying task config vm is no longer supported"));
+        }
+
+        if (config.Pool != null) {
```
Comment (stishkin) [inline src/ApiService/ApiService/onefuzzlib/Config.cs:297]: config.Pool == null

File: src/ApiService/ApiService/onefuzzlib/Config.cs [hunk 3]
```diff
@@ -257,4 +264,257 @@ await _containers.GetContainerSasUrl(x.Item2.Name, StorageType.Corpus, ConvertPe
...
+    }
+
+    private async Task<ResultVoid<TaskConfigError>> CheckTargetExe(TaskConfig config, TaskDefinition definition) {
+        if (config.Task.TargetExe == null) {
+            if (definition.Features.Contains(TaskFeature.TargetExe)) {
+                return ResultVoid<TaskConfigError>.Error(new TaskConfigError("missing target_exe"));
+            }
+
+            if (definition.Features.Contains(TaskFeature.TargetExeOptional)) {
+                return ResultVoid<TaskConfigError>.Ok();
```
Comment (stishkin) [inline src/ApiService/ApiService/onefuzzlib/Config.cs:359]: is there value to have this "if" statement ? the return from the function is the same as from within the if statement

**Review by stishkin — APPROVED** [review]

File: src/ApiService/ApiService/onefuzzlib/Config.cs (hunk 3 above)
Comment (chkeita) [reply src/ApiService/ApiService/onefuzzlib/Config.cs:359]: yeah i thought it was weird too but I kept it to match the [original code ](https://github.com/chkeita/onefuzz/blob/b2399c45710b52160b994e429fe3393a292da8be/src/api-service/__app__/onefuzzlib/tasks/config.py#L109)

**Pull request merged**